frontend/app/routes/$lang+/_protected+/alerts+/subscribe+/confirm.tsx

```diff
@@ -0,0 +1,109 @@
+
+export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  featureEnabled('email-alerts');
+
+  const instrumentationService = getInstrumentationService();
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  const csrfToken = String(session.get('csrfToken'));
+  const meta = { title: t('gcweb:meta.title.template', { title: t('alerts:confirm.page-title') }) };
+
+  const personalInformationRouteHelper = getPersonalInformationRouteHelpers();
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation: PersonalInfo = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+  const preferredLanguage = personalInformation.preferredLanguageId ? await getLookupService().getPreferredLanguage(personalInformation.preferredLanguageId) : undefined;
+  const clickedButton = '';
+  instrumentationService.countHttpStatus('alerts.confirm', 302);
+  return json({ csrfToken, meta, personalInformation, preferredLanguage, clickedButton });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const personalInformationRouteHelper = getPersonalInformationRouteHelpers();
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation: PersonalInfo = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+
+  const formData = await request.formData();
+
+  if (formData.get('new-code')) {
+    //TODO EMAIL THE CODE TO THE CLIENTS ADDRESS
+    personalInformation.emailAddress;
+  }
+
+  if (formData.get('submit')) {
+    //TODO VALIDATE THE CONFIRMATION CODE
+  }
+
+  return '';
+}
+
+export default function ConfirmSubscription() {
+  const { i18n, t } = useTranslation(handle.i18nNamespaces);
+  const { preferredLanguage } = useLoaderData<typeof loader>();
+  const params = useParams();
+  const fetcher = useFetcher<typeof action>();
+  const userOrigin = useUserOrigin();
+
+  return (
+    <>
+      <fetcher.Form className="max-w-prose" method="post" noValidate>
+        <div className="mb-8 space-y-6">
+          <p id="confirmation-information" className="mb-4">
+            {t('alerts:confirm.confirmation-information-text')}
+          </p>
+          <p id="confirmation-completed" className="mb-4">
+            {t('alerts:confirm.confirmation-completed-text')}
+          </p>
+          <div className="grid gap-12 md:grid-cols-2">
+            <p id="confirmation-language" className="mb-4">
+              {t('alerts:confirm.confirmation-selected-language')}
+            </p>
+            <p>{preferredLanguage ? getNameByLanguage(i18n.language, preferredLanguage) : t('alerts:confirm.no-preferred-language-on-file')}</p>
+          </div>
```

Comment:
Surround this with a `<ContextualAlert>` component to make the content stand out (similar to how it's done in the MSCA screenshot attached in the user story).

Reply:
Contextual alert as been put in place